unit flake: reflector_test.go:384: expected exiting from err expected resync reached, got: <nil>

In per-PR build:
```
--- FAIL: TestReflectorResync (0.01s)
	reflector_test.go:384: expected exiting from err expected resync reached, got: <nil>
```
https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/26054/kubernetes-pull-test-unit-integration/27306/

cc @kubernetes/sig-api-machinery @wojtek-t @lavalamp

--- a/pkg/client/cache/reflector_test.go
+++ b/pkg/client/cache/reflector_test.go
@@ -379,9 +379,9 @@ func TestReflectorResync(t *testing.T) {
 	}
 	resyncPeriod := 1 * time.Millisecond
 	r := NewReflector(lw, &api.Pod{}, s, resyncPeriod)
-	err := r.ListAndWatch(stopCh)
-	if err != rerr {
-		t.Errorf("expected exiting from err %v, got: %v", rerr, err)
+	if err := r.ListAndWatch(stopCh); err != nil {
+		// error from Resync is not propaged up to here.
+		t.Errorf("expected error %v", rerr, err)
 	}
 	if iteration != 2 {
 		t.Errorf("exactly 2 iterations were expected, got: %v", iteration)

